<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>EditorWindow</class>
 <widget class="QMainWindow" name="EditorWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1270</width>
    <height>835</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>The ORF 2017!</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <layout class="QVBoxLayout" name="ToolBar" stretch="0">
      <item>
       <layout class="QHBoxLayout" name="Tools">
        <property name="sizeConstraint">
         <enum>QLayout::SetNoConstraint</enum>
        </property>
        <item>
         <widget class="QPushButton" name="AxesButton">
          <property name="text">
           <string>Axis On/Off</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="WireframeButton">
          <property name="text">
           <string>Solid mesh / wireframe</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="PerspectiveButton">
          <property name="text">
           <string>Orthographic</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="ColorButton">
          <property name="text">
           <string>Set background color</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="ToggleCamera">
          <property name="text">
           <string>Editor/Game Camera</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QPushButton" name="SaveScene">
      <property name="text">
       <string>Save Scene</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="LoadScene">
      <property name="text">
       <string>Load Scene</string>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="Main">
      <item>
       <layout class="QVBoxLayout" name="HierarchyLayout">
        <item>
         <widget class="QLabel" name="HierarchyLabel">
          <property name="text">
           <string>Hierarchy</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="Cube">
          <property name="minimumSize">
           <size>
            <width>75</width>
            <height>23</height>
           </size>
          </property>
          <property name="text">
           <string>Cube</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="Ball">
          <property name="text">
           <string>Ball</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="Plane">
          <property name="text">
           <string>Plane</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="ReadObjFile">
          <property name="text">
           <string>Read from File</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="FilePathInput"/>
        </item>
        <item>
         <widget class="QListWidget" name="HierarchyList"/>
        </item>
       </layout>
      </item>
      <item>
       <widget class="GameEngine" name="RenderWindow">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>100</horstretch>
          <verstretch>100</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="mouseTracking">
         <bool>false</bool>
        </property>
        <property name="focusPolicy">
         <enum>Qt::ClickFocus</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="Transform">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>200</width>
          <height>200</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>200</width>
          <height>250</height>
         </size>
        </property>
        <property name="title">
         <string>Transform</string>
        </property>
        <widget class="QLineEdit" name="InspectPosX">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>40</y>
           <width>40</width>
           <height>20</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="InspectPosY">
         <property name="geometry">
          <rect>
           <x>80</x>
           <y>40</y>
           <width>40</width>
           <height>20</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="InspectPosZ">
         <property name="geometry">
          <rect>
           <x>150</x>
           <y>40</y>
           <width>40</width>
           <height>20</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="InspectRotZ">
         <property name="geometry">
          <rect>
           <x>150</x>
           <y>80</y>
           <width>40</width>
           <height>20</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="InspectRotY">
         <property name="geometry">
          <rect>
           <x>80</x>
           <y>80</y>
           <width>40</width>
           <height>20</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="InspectRotX">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>80</y>
           <width>40</width>
           <height>20</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="InspectScaZ">
         <property name="geometry">
          <rect>
           <x>150</x>
           <y>120</y>
           <width>40</width>
           <height>20</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="InspectScaY">
         <property name="geometry">
          <rect>
           <x>80</x>
           <y>120</y>
           <width>40</width>
           <height>20</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="InspectScaX">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>120</y>
           <width>40</width>
           <height>20</height>
          </rect>
         </property>
        </widget>
        <widget class="QLabel" name="label">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>40</y>
           <width>16</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>X</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_2">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>80</y>
           <width>16</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>X</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_3">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>120</y>
           <width>16</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>X</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_4">
         <property name="geometry">
          <rect>
           <x>130</x>
           <y>80</y>
           <width>16</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Z</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_5">
         <property name="geometry">
          <rect>
           <x>130</x>
           <y>40</y>
           <width>16</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Z</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_6">
         <property name="geometry">
          <rect>
           <x>130</x>
           <y>120</y>
           <width>16</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Z</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_7">
         <property name="geometry">
          <rect>
           <x>70</x>
           <y>80</y>
           <width>16</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Y</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_8">
         <property name="geometry">
          <rect>
           <x>70</x>
           <y>40</y>
           <width>16</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Y</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_9">
         <property name="geometry">
          <rect>
           <x>70</x>
           <y>120</y>
           <width>16</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Y</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_10">
         <property name="geometry">
          <rect>
           <x>80</x>
           <y>14</y>
           <width>47</width>
           <height>20</height>
          </rect>
         </property>
         <property name="text">
          <string>Position</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_11">
         <property name="geometry">
          <rect>
           <x>80</x>
           <y>60</y>
           <width>47</width>
           <height>13</height>
          </rect>
         </property>
         <property name="text">
          <string>Rotation</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_12">
         <property name="geometry">
          <rect>
           <x>90</x>
           <y>100</y>
           <width>47</width>
           <height>13</height>
          </rect>
         </property>
         <property name="text">
          <string>Scale</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="ObjName">
         <property name="geometry">
          <rect>
           <x>60</x>
           <y>0</y>
           <width>133</width>
           <height>20</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="lineEdit">
         <property name="geometry">
          <rect>
           <x>70</x>
           <y>160</y>
           <width>113</width>
           <height>20</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="lineEdit_2">
         <property name="geometry">
          <rect>
           <x>70</x>
           <y>200</y>
           <width>113</width>
           <height>20</height>
          </rect>
         </property>
        </widget>
        <widget class="QLabel" name="label_25">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>200</y>
           <width>71</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Vertex Count</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_26">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>160</y>
           <width>47</width>
           <height>13</height>
          </rect>
         </property>
         <property name="text">
          <string>FPS</string>
         </property>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QLabel" name="Infobox">
      <property name="text">
       <string>X</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1270</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionQuit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionLoad_Mesh">
   <property name="text">
    <string>Load Mesh</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
  <action name="actionSave_Scene">
   <property name="text">
    <string>Save Scene</string>
   </property>
  </action>
  <action name="actionLoad_Scene">
   <property name="text">
    <string>Load Scene</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GameEngine</class>
   <extends>QOpenGLWidget</extends>
   <header>gameengine.h</header>
   <slots>
    <signal>initHierarchy()</signal>
    <slot>axisOnOff()</slot>
    <slot>loadMesh()</slot>
    <slot>wireFrameOnOff()</slot>
    <slot>setCameraColor()</slot>
    <slot>orthographicOnOff()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>RenderWindow</tabstop>
  <tabstop>AxesButton</tabstop>
  <tabstop>WireframeButton</tabstop>
  <tabstop>PerspectiveButton</tabstop>
  <tabstop>ColorButton</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>PerspectiveButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>RenderWindow</receiver>
   <slot>orthographicOnOff()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>688</x>
     <y>45</y>
    </hint>
    <hint type="destinationlabel">
     <x>688</x>
     <y>86</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>AxesButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>RenderWindow</receiver>
   <slot>axisOnOff()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>230</x>
     <y>49</y>
    </hint>
    <hint type="destinationlabel">
     <x>523</x>
     <y>216</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>WireframeButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>RenderWindow</receiver>
   <slot>wireFrameOnOff()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>479</x>
     <y>53</y>
    </hint>
    <hint type="destinationlabel">
     <x>535</x>
     <y>92</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
